### PR TITLE
1831 remove windows executable functionality

### LINKED ISSUE #1831: Round Number Delay When Starting Under WSL
Multiple WSL users have reported issues with a freeze on startup, always a round number: 20s, 100s, 60s

Git enabled 20s: #1719

Git disabled 60s/100s: https://github.com/nvim-tree/nvim-tree.lua/issues/549#issuecomment-1354354894

### PR DESCRIPTION
**Pull request by alex-courtis**
fixes #1831

### CODE REVIEW

File: lua/nvim-tree/utils.lua [hunk 1]
```diff
@@ -152,63 +155,6 @@ function M.get_nodes_by_line(nodes_all, line_start)
...
-    local handle = io.popen 'cmd.exe /c "echo %PATHEXT%" 2>/dev/null'
-    if handle then
-      vim.env.PATHEXT = handle:read "*a"
-      handle:close()
-    end
-  end
-
-  return M.is_windows_exe(ext)
-end
-
```
Comment (alex-courtis) [inline lua/nvim-tree/utils.lua:211]: Feels good.

**Review by gegoune — APPROVED** [review]
Well done with factoring out flags to utils. Have seen those checks in code all over place before and it was such an eyesore.

**Comment by alex-courtis** [comment]
> Well done with factoring out flags to utils. Have seen those checks in code all over place before and it was such an eyesore.

There are only a few places they are actually used.

I was suprised to learn that trash doesn't work on macOS or windows...

**Comment by gegoune** [comment]
Trash itself doesn't but there is another tool, it might even be shipped with macOS, but I never had a need to use it so can't recall its name.

**Pull request merged**